motionplan/cBiRRT.go

```diff
@@ -110,115 +105,167 @@ func NewCBiRRTMotionPlannerWithSeed(frame referenceframe.Frame, nCPU int, seed *
 func (mp *cBiRRTMotionPlanner) Plan(ctx context.Context,
 	goal spatialmath.Pose,
 	seed []referenceframe.Input,
-	planOpts *PlannerOptions,
+	planOpts *plannerOptions,
 ) ([][]referenceframe.Input, error) {
 	if planOpts == nil {
-		planOpts = NewBasicPlannerOptions()
+		planOpts = newBasicPlannerOptions()
 	}
-	solutionChan := make(chan *planReturn, 1)
+	solutionChan := make(chan *rrtPlanReturn, 1)
 	utils.PanicCapturingGo(func() {
-		mp.planRunner(ctx, goal, seed, planOpts, nil, solutionChan)
+		mp.rrtBackgroundRunner(ctx, goal, seed, planOpts, initRRTMaps(), nil, solutionChan)
 	})
 	select {
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	case plan := <-solutionChan:
-		return plan.toInputs(), plan.err
+		return plan.ToInputs(), plan.err
 	}
 }
 
-// planRunner will execute the plan. When Plan() is called, it will call planRunner in a separate thread and wait for the results.
-// Separating this allows other things to call planRunner in parallel while also enabling the thread-agnostic Plan to be accessible.
-func (mp *cBiRRTMotionPlanner) planRunner(
+// rrtBackgroundRunner will execute the plan. Plan() will call RRTBackgroundRunner in a separate thread and wait for results.
+// Separating this allows other things to call RRTBackgroundRunner in parallel allowing the thread-agnostic Plan to be accessible.
+func (mp *cBiRRTMotionPlanner) rrtBackgroundRunner(
 	ctx context.Context,
 	goal spatialmath.Pose,
 	seed []referenceframe.Input,
-	planOpts *PlannerOptions,
+	planOpts *plannerOptions,
+	rm *rrtMaps,
 	endpointPreview chan node,
-	solutionChan chan *planReturn,
+	solutionChan chan *rrtPlanReturn,
 ) {
 	defer close(solutionChan)
 
 	// setup planner options
 	if planOpts == nil {
-		solutionChan <- &planReturn{err: errNoPlannerOptions}
+		solutionChan <- &rrtPlanReturn{err: errNoPlannerOptions}
 		return
 	}
 	algOpts, err := newCbirrtOptions(planOpts, mp.frame)
 	if err != nil {
-		solutionChan <- &planReturn{err: err}
+		solutionChan <- &rrtPlanReturn{err: err}
 		return
 	}
 
+	// initialize maps
+	corners := map[node]bool{}
```

Comment:
[q] do we need to consider how this plays with passed in maps?

Reply:
Added contents to corners